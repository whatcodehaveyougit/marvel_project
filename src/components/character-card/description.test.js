import { render } from '@testing-library/react';
import character2 from './testData/character2.json';
import React, { useRef } from 'react';
import Description from './description';


// Remove the unnecessary console.log statement
// React.useRef = () => ({ current: { clientHeight: 100, scrollHeight: 200 } });

describe('Description', () => {

  it('renders correctly with another character and useRef being mocked', () => {
    // reactMock.useRef = { current: { clientHeight: 100, scrollHeight: 200 } };
    // const useRefSig = jest.fn();
    jest.mock('react', () => {
      // const originReact = jest.requireActual('react');
      return {
        // ...originReact,
        'hello': 'world',
        'useRef': { current: { clientHeight: 100, scrollHeight: 200 } }, // Not working
      };
    });

    const { asFragment } = render(
        <Description character={character2} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

});


// Problem

// Cannot mock useRef in the test file and the HTML components are not rendered with
// All the attributes and values as normal so it does't go through the whole if statement
// So doesn't test the whole file
// I think it is useRef that I need to mock but not 100% sure.,.