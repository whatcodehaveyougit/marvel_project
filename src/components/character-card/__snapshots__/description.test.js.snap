// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description renders correctly with another character and useRef being mocked 1`] = `
<DocumentFragment>
  <div
    class="container-1017100 max-lines"
  >
    <div
      class="text-gray-500"
    >
      <p
        id="contentdiv-1017100"
      >
        Rick Jones has been Hulk's best bud since day one, but now he's more than a friend...he's a teammate! Transformed by a Gamma energy explosion, A-Bomb's thick, armored skin is just as strong and powerful as it is blue. And when he curls into action, he uses it like a giant bowling ball of destruction! 
      </p>
    </div>
  </div>
</DocumentFragment>
`;
